lib/modules/manager/hermit/artifacts.ts

```diff
@@ -0,0 +1,259 @@
+  const added = await pMap(
+    [...hermitChanges.created, ...hermitChanges.not_added],
+    async (path: string): Promise<UpdateArtifactsResult> => {
+      const contents = await getContent(path);
+
+      return getAddResult(path, contents);
+    }
+  );
+
+  const deleted = hermitChanges.deleted.map(getDeleteResult);
+
+  const modified = await pMap(
+    hermitChanges.modified,
+    async (path: string): Promise<UpdateArtifactsResult[]> => {
+      const contents = await getContent(path);
+      return [
+        getDeleteResult(path), // delete existing link
+        getAddResult(path, contents), // add a new link
+      ];
+    }
+  );
+
+  const renamed = await pMap(
+    hermitChanges.renamed,
+    async (renamed): Promise<UpdateArtifactsResult[]> => {
+      const from = renamed.from;
+      const to = renamed.to;
+      const toContents = await getContent(to);
+
+      return [getDeleteResult(from), getAddResult(to, toContents)];
+    }
+  );
+
+  return [
+    // rename will need to go first, because
+    // it needs to create the new link for the new version
+    // for the modified links to use
+    ...flattern(renamed),
+    ...flattern(modified),
+    ...added,
+    ...deleted,
+  ];
+}
+
+/**
+ * getHermitPackage returns the hermit package for running the hermit install
+ */
+function getHermitPackage(name: string, version: string): string {
+  return `${name}-${version}`;
+}
+
+/**
+ * getHermitPackageReferenceFile returns the hermit package reference
+ * file with the given package name and version
+ */
+function getHermitPackageReferenceFile(
+  pkgName: string,
+  version: string
+): string {
+  return `bin/.${getHermitPackage(pkgName, version)}.pkg`;
+}
+
+/**
+ * updateHermitPackage runs hermit install for the given package
+ */
+async function updateHermitPackage(
+  pkg: HermitPackageDependency
+): Promise<UpdateHermitResult | void> {
+  logger.trace({ pkg }, `hermit.updateHermitPackage()`);
+  if (!pkg.depName || !pkg.currentVersion || !pkg.newValue) {
+    logger.error(
+      {
+        depName: pkg.depName,
+        currentVersion: pkg.currentVersion,
+        newValue: pkg.newValue,
+      },
+      'missing package update information'
+    );
+    throw {
+      stderr: `invalid package to update`,
+    };
+  }
+  const depName = pkg.depName;
+  const currentVersion = pkg.currentVersion;
+  const newValue = pkg.newValue;
+  const from = getHermitPackageReferenceFile(depName, currentVersion);
+  const to = getHermitPackageReferenceFile(depName, newValue);
+  const fromPackage = getHermitPackage(depName, currentVersion);
+  const toPackage = getHermitPackage(depName, newValue);
+  const execOptions: ExecOptions = {
+    extraEnv: {
+      // hermit takes GITHUB_TOKEN from the following environment variable to perform Github release download
+      // HERMIT_GITHUB_TOKEN has higher priority than GITHUB_TOKEN
+      HERMIT_GITHUB_TOKEN: process.env.HERMIT_GITHUB_TOKEN,
+      GITHUB_TOKEN: process.env.GITHUB_TOKEN,
```

Comment:
The token access required for `hermit install` will be wider than the datasource. Host rules might be a bit difficult. 

However if feeding through process.env is not preferred here. It could still come from the `customEnvironmentVariables`. I will document it instead.